sdk/metric/sdk.go

```diff
@@ -182,13 +182,9 @@ func (a *asyncInstrument) observe(num number.Number, labels *label.Set) {
 func (a *asyncInstrument) getRecorder(labels *label.Set) export.Aggregator {
 	lrec, ok := a.recorders[labels.Equivalent()]
 	if ok {
-		if lrec.observedEpoch == a.meter.currentEpoch {
-			// last value wins for Observers, so if we see the same labels
-			// in the current epoch, we replace the old recorder
-			a.meter.processor.AggregatorFor(&a.descriptor, &lrec.observed)
-		} else {
-			lrec.observedEpoch = a.meter.currentEpoch
-		}
+		// Note: SynchronizedMove(nil) can't return an error
+		_ = lrec.observed.SynchronizedMove(nil, &a.descriptor)
```

Comment:
Note: This is the line where this bug was fixed.  The `if` branch here was doing a reset, but the `else` branch was not. The code was originally correct and then a bunch of things changed around it, introducing the bug (@krnowak take note!). Specifically, I moved from the Accumulator keeping two Aggregators to keeping one Aggregator, as part of introducing a more-complex Processor component.

The new Processor code was heavily tested, but the former Accumulator wasn't tested enough for the Observer instruments, particularly with multiple rounds. So I had introduced a bug ages ago in the move to a single Aggregator per Accumulator record. We could've fixed this code by calling 
```
a.meter.processor.AggregatorFor(&a.descriptor, &lrec.observed)
```
unconditionally (the `if`-branch), but that's a wasted allocation. Introducing support for resetting the Aggregator fixes the bug w/o the wasted allocation.

Reply:
I'm referring to this PR where I think the bug starts. https://github.com/open-telemetry/opentelemetry-go/pull/812